### ISSUE TITLE
Unable to view folder contents when importing collection with folders on Mac

### ISSUE THREAD

**Issue by BigSauce** [a85/POSTMan-Chrome-Extension#946]
Google Chrome:  Version 40.0.2214.115 (64-bit)
Postman - REST Client (Packaged App): Version 2.0.1 (packaged)
Mac OS X Yosemite 10.10.2

Whenever I import a collection that contains folders, the folder arrows rotate, but nothing shows in the folders.  I know the folders exist, as I am able to view the folders on a PC.  I am thus unable to use Postman for my collections until this is resolved.

Hopefully this can be fixed soon.  Thanks!

![image](https://cloud.githubusercontent.com/assets/1718470/6467455/4ea11124-c19c-11e4-90c7-50dab10f3566.png)

**Comment by abhijitkane** [comment]
You can view the folders' requests on a PC with version 2.0.1 of Postman?

Is this happening with all collections that you import?

**Comment by BigSauce** [comment]
Correct.  I can view it on a PC but not on my Mac.  I've tried re-installing Postman and restarting Chrome multiple times.

I only have one collection on hand to test.  Do you have any collections I could attempt to import?

**Comment by abhijitkane** [comment]
Can you mail the collection file to help@getpostman.com?

You can try importing this - https://www.getpostman.com/collections/43ec0b8cc4834bf66fd7
It has one collection, one folder, and one request in the folder.

**Comment by BigSauce** [comment]
Unfortunately, I am not allowed to share the collection external to my company (I know that makes it more difficult for you guys to debug), but I did import the collection you provided, and I was able to view the group in that.  Makes me wonder if the json format for collections have changed recently.  Have they?

![image](https://cloud.githubusercontent.com/assets/1718470/6468785/88c0c238-c1a5-11e4-8d29-b01c8b3e8e4b.png)

**Comment by abhijitkane** [comment]
No. You can export the collection as a JSON file, and change the URLs / headers / body parameters.
If the format were a problem, you wouldn't be able to view it on a PC either.

One thing that you can try - enable the developer tools (http://blog.getpostman.com/index.php/2014/01/27/enabling-chrome-developer-tools-inside-postman/)

And check for errors in the console after you import your collection.

**Comment by BigSauce** [comment]
Ok I enabled the developer tools per your advice, and nothing is printed in the console when importing the collection.  The following screenshot, however, is what shows in the console upon launch of Postman (packaged app) from Chrome:

![image](https://cloud.githubusercontent.com/assets/1718470/6469260/dfc10202-c1a8-11e4-888f-ebebd72a4055.png)

![image](https://cloud.githubusercontent.com/assets/1718470/6469274/010d35e8-c1a9-11e4-939d-18e790006cef.png)

In case you want to see info about the window.onbeforeunload error:
![image](https://cloud.githubusercontent.com/assets/1718470/6469266/f38304f2-c1a8-11e4-9f92-eb988e4014d4.png)

**Comment by BigSauce** [comment]
Is there a verbose logging mode I can utilize?

**Comment by abhijitkane** [comment]
Hm...those are unrelated issues. And no, that's all the logging there is. 

You could try duplicating the collection, and then exporting and re-importing it. Inconsistencies in the current collection might be solved by duplicating it. (There's a duplicate icon when you hover over the collection name)

**Comment by BigSauce** [comment]
Duplicating the collection caused the requests to be listed after the groups rather than embedded inside of them.  Same output was generated when exporting and re-importing the duplicated collection.

**Comment by BigSauce** [comment]
![image](https://cloud.githubusercontent.com/assets/1718470/6470363/4167031e-c1b1-11e4-9569-60c307fcf8d0.png)

**Comment by abhijitkane** [comment]
If you move the requests into the correct folders (in the duplicated collection), do you still see the problem?

Not sure what's causing this, though - we'll continue investigating.

**Comment by dhoyt** [comment]
Hello ...

I work with BigSauce and was able to isolate the problem and fix it. The JSON file coming from our server did not have an "order": [] in the root 'collection'. I made a change to PostmanCollection.java to create this array and all seems to be working correctly now.

There had to be something in the latest update to the POSTMAN app that was sensitive to this setting and causing the condition we were encountering.

Is there any documentation that I can get my hands on that describes all the JSON settings/objects that the POSTMAN front end uses when importing ?

Is there a setting in the JSON file that we can use to default the opening/closing of the description when an 'request' is selected on the Front End. Having the description open, by default, is painful if one is using the app all day long to create regression test collections.

Thanks for your quick attention to this matter.

**Comment by abhijitkane** [comment]
We don't have any documentation for the format yet. It's only meant to be created in Postman, and consumed in Postman and Newman.

Each JSON file is a collection object. The collection.requests array contains all requests in the collection (whether in the collection directly, or in folders). The collection.folders array contains folders. Each folder has an order array which contains the IDs of requests in that folder. The collection.order array contains IDs of requests that are directly in the collection. (Each element of collection.folders[x].order, and collection.order must be an ID of atleast one member of collection.requests)

**Comment by dhoyt** [comment]
But you do support 'importing' from a URL ? ... which is what we use to get our master API set into POSTMAN. It was the importing of the JSON file from our server where the problem arose. The supportive POSTMAN code on our server, that generates the 'collection' JSON file, was not handling the notion of a collection.order array, which I had to add to make to get our import to work correctly. 

The example file that you sent us shows some settings that we don't generate. That was the reason that I asked if there was any documentation. I can make changes to the 'import' file as it is being created, on our server, and thought that maybe I could take advantage of some setting(s) that POSTMAN may have that would make our API endpoints more informational/robust and easier to work with. 

I was hoping that my second question, in my previous post, was one of those settings.

**Comment by abhijitkane** [comment]
I'll work on a template for a collection file, with all the Postman-specific fields filled in. I'm guessing you only generate the HTTP verb, url, headers, and the optional request body.

We're also working on comprehensive documentation for the format, but we don't have a timeline on that yet.

**Comment by a85** [comment]
@dhoyt Actually while documenting this format we found several issues and that's why we we started working on a new format. It should be out soon with supporting libraries that will make it easier to generate collections and issues like the one you mentioned would not exist. We'll post an update here and on our blog (http://blog.getpostman.com). We are quite close to finalizing that and will open up to comments asap.
